airbyte-db/src/main/java/io/airbyte/db/JdbcUtils.java

```diff
@@ -0,0 +1,231 @@
+          return true;
+        } catch (SQLException e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+    }, false);
+  }
+
+  /**
+   * Collect each record of a ResultSet into a list of JsonNode.
+   *
+   * @param resultSet the result set
+   * @return List of JsonNode.
+   * @throws SQLException exceptions throws when parsing the ResultSet.
+   */
+  public static List<JsonNode> resultSetToJson(ResultSet resultSet) throws SQLException {
+    return mapResultSet(resultSet, JdbcUtils::getJsonForRow).collect(Collectors.toList());
+  }
+
+  /**
+   * Stream the each record of a ResultSet lazily.
+   *
+   * @param preparedStatement that needs to be executed lazily.
+   * @return Stream of JsonNode.
+   * @throws SQLException exceptions throws when parsing the ResultSet.
+   */
+  public static <T> Stream<T> fetchStream(PreparedStatement preparedStatement, CheckedFunction<ResultSet, T, SQLException> mapper)
+      throws SQLException {
+    return Stream.of(1).flatMap(i -> {
+      try {
+        return mapResultSet(preparedStatement.executeQuery(), mapper);
+      } catch (SQLException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  /**
+   * Stream the each record of a ResultSet as JsonNode lazily.
+   *
+   * @param preparedStatement that needs to be executed lazily.
+   * @return Stream of JsonNode.
+   * @throws SQLException exceptions throws when parsing the ResultSet.
+   */
+  public static Stream<JsonNode> fetchStreamAsJson(PreparedStatement preparedStatement) throws SQLException {
+    return fetchStream(preparedStatement, JdbcUtils::getJsonForRow);
+  }
+
+  /**
+   * Iterator of each record of a ResultSet as JsonNode.
+   *
+   * @param resultSet the result set
+   * @return Stream of JsonNode.
+   * @throws SQLException exceptions throws when parsing the ResultSet.
+   */
+  public static Iterator<JsonNode> resultSetToJsonIterator(ResultSet resultSet) throws SQLException {
+    return mapResultSet(resultSet, JdbcUtils::getJsonForRow).iterator();
+  }
+
+  public static JsonNode getJsonForRow(ResultSet r) throws SQLException {
+    // the first call communicates with the database. after that the result is cached.
+    final int columnCount = r.getMetaData().getColumnCount();
+    final ObjectNode jsonNode = (ObjectNode) Jsons.jsonNode(Collections.emptyMap());
+
+    for (int i = 1; i <= columnCount; i++) {
+      // attempt to access the column. this allows us to know if it is null before we do type-specific
+      // parsing. if it is null, we can move on. while awkward, this seems to be the agreed upon way of
+      // checking for null values with jdbc.
+      r.getObject(i);
+      if (r.wasNull()) {
+        continue;
+      }
+
+      // convert to java types that will convert into reasonable json.
+      jdbcToJson(jsonNode, r, i);
+    }
+
+    return jsonNode;
+  }
+
+  public static void jdbcToJson(ObjectNode o, ResultSet r, int i) throws SQLException {
```

Comment:
`jdbcColumnToJson`
NIT: can you make ` o` the 3rd argument? (input, output) and it matches the function name as well

Reply:
setJsonField(o, r, i)